### PR TITLE
Rate-limit login form #381

### LINKED ISSUE #381: Rate-limit login form
Right now (AFAIK) we don't rate-limit our login form, which means that you could feasibly just brute-force a password and login. We should implement some kind of rate-limiting that makes you cool off for 15 minutes after a few failed logins. 

I'm sure this is something devise does, either with an extension or with something else. 

(For all I know, this is already turned on! but we should confirm that's the case if so.)

### PR DESCRIPTION
**Pull request by ashlynnpai**
I rule and have completed some work on Case Manager that's ready for review!

(brief, plain english overview of your changes here)

This pull request makes the following changes:
* configure lockable in initializers/devise.rb for account lock for 15 mins after 5 failed logins
* enable lockable module in user.rb
* enable lockable fields in user.rb

It relates to the following issue #s: 
* Fixes #381

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -8,9 +8,9 @@ class User
           :recoverable,
           :rememberable,
           :trackable,
-          :validatable
+          :validatable,
+          :lockable
```
Comment (colinxfleming) [inline app/models/user.rb]: 💯

File: bin/rails [hunk 2]
```diff
@@ -1,8 +1,4 @@
 #!/usr/bin/env ruby
-begin
```
Comment (colinxfleming) [inline bin/rails]: Can you put this block back in? Would like to keep `spring` in if we can!

File: bin/rake [hunk 3]
```diff
@@ -1,8 +1,4 @@
 #!/usr/bin/env ruby
-begin
```
Comment (colinxfleming) [inline bin/rake]: Same deal -- can you pop `spring` back in?

File: config/initializers/devise.rb [hunk 4]
```diff
@@ -158,7 +158,7 @@
   # Defines which strategy will be used to lock an account.
   # :failed_attempts = Locks an account after a number of failed attempts to sign in.
   # :none            = No lock strategy. You should handle locking by yourself.
-  # config.lock_strategy = :failed_attempts
+  config.lock_strategy = :failed_attempts
```
Comment (colinxfleming) [inline config/initializers/devise.rb]: yussss

File: config/puma.rb [hunk 5]
```diff
@@ -11,5 +11,7 @@
 on_worker_boot do
   # Worker specific setup for Rails 4.1+
   # See: https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server#on-worker-boot
-  ActiveRecord::Base.establish_connection
+  ActiveSupport.on_load(:active_record) do
```
Comment (colinxfleming) [inline config/puma.rb]: You can probably revert this change too -- I switched us back to `unicorn` because puma was giving me some unrelated guff.

**Comment by colinxfleming** [comment]
Love this, thank you so much for your work here @ashlynnpai ! Happy you got to enjoy Mongo's best feature, beating the tar out of the models on the fly. 💯 

Had a few comments, and it looks like there are some merge conflicts to resolve, but I think after that we can declare victory and merge this up!

**Comment by ashlynnpai** [comment]
Please let me know if I cleaned up the merge conflicts okay.

**Comment by colinxfleming** [comment]
Looks good! Thanks for your quick action @ashlynnpai !

**Comment by colinxfleming** [comment]
(The CI is failing because of a lousy integration test I wrote; this doesn't affect actual logging calls functionality.)

**Comment by colinxfleming** [comment]
@ashlynnpai we have a couple of similar issues to this (wrenching on devise configs) in #388 (disabling `:rememberable`) and #379 (turning on `:timeoutable`). Do you want to take a stab at those to play around with devise a little more?

**Comment by ashlynnpai** [comment]
I'll take a look.

**Pull request merged**